tests/unit/expiry/HgvTrlMostRecentExpiryStrategy.unitTest.ts

```diff
@@ -0,0 +1,147 @@
+import { cloneDeep } from "lodash";
+import testResults from "../../resources/test-results.json";
+import { TestTypeForExpiry } from "../../../src/models/TestTypeforExpiry";
+import { DateProvider } from "../../../src/handlers/expiry/providers/DateProvider";
+import { TestType } from "../../../src/models/ITestResult";
+import { VEHICLE_TYPE } from "../../../src/assets/Enums";
+import { HgvTrlMostRecentExpiryStrategy } from "../../../src/handlers/expiry/strategies/HgvTrlMostRecentExpiryStrategy";
+
+describe("HgvTrlMostRecentExpiryStrategy", () => {
+  let testResultsMockDB: any;
+  let hgvTrlMostRecentExpiryStrategy: HgvTrlMostRecentExpiryStrategy;
+
+  beforeEach(() => {
+    testResultsMockDB = cloneDeep(testResults);
+  });
+
+  afterEach(() => {
+    // reset date to current date
+    hgvTrlMostRecentExpiryStrategy.dateProvider.setTestDate(new Date());
+  });
+
+  context("for hgv vehicle type", () => {
+    describe("with missing expiry date", () => {
+      it("should set the Expiry last day of month + 1 year of testDate", () => {
```

Comment:
We should be testing more than the last day of the month.  We should check that the first day of the month, middle.  We should check that on leap years we wind forward to the 29th.  If this used a test.each table - we could quickly and easily test the start, middle and end of every month in a few dozen lines.

The recent expiry is passed as the epoch date - shouldn't this be empty/undefined as that is what the rule is.  Also the regn date is passed, we should test for this being present and absent (think the rules are the same)

Reply:
The tests are replaced with jest table approach and the test for recent Expiry to be epoch or undefined is invalid in regards to `HgvTrlMostRecentExpiryStrategy` as this strategy requires a valid recent expiry. This test case is removed.